src/openms/include/OpenMS/DATASTRUCTURES/StringUtils.h

```diff
@@ -60,7 +60,16 @@ namespace OpenMS
     class BK_PrecPolicy : public boost::spirit::karma::real_policies<T>
     {
     public:
-      static unsigned int precision(T) { return writtenDigits<T>(T()); }
+        static unsigned int precision(T n)
+        {
+          // max_digits10 specifies the max digits needed to uniquely read floating points
+          // back in. The precision member only controls the fraction part.
+          // -> if fixed point, subtract the number of digits of the integer part
+          // -> if scientific, subtract one
+          return BK_PrecPolicy<T>::floatfield(n) ?
+            std::numeric_limits<T>::max_digits10 - (floor(log10(n)) + 1) : 
+            std::numeric_limits<T>::max_digits10 - 1;
```

Comment:
why subtract one?
From the boost docu:
The extra two or three least-significant digits are 'noisy' and may be junk, but if you want to 'round-trip' - printing a value out as a decimal digit string and reading it back in - (most commonly during serialization and de-serialization) you must use os.precision(std::numeric_limits<T>::max_digits10)

Reply:
for scientifc notation only.